### PR TITLE
add: list_availability_zones to shares

### PR DESCRIPTION
**Pull request by andypf**
@Carthaca: Please take a look

### CODE REVIEW

File: lib/fog/shared_file_system/openstack.rb [hunk 1]
```diff
@@ -179,6 +182,14 @@ def self.data
                   "name"  => "snapshot_My_share"
                 }
               ],
+              :availability_zones => [
+                  {
+                      "name": "nova",
```
Comment (Carthaca) [inline lib/fog/shared_file_system/openstack.rb]: please use hash rocket syntax

File: lib/fog/shared_file_system/openstack/requests/list_availability_zones.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module Fog
+  module SharedFileSystem
+    class OpenStack
+      class Real
+        def list_availability_zones()
+          request(
+            :expects => 200,
+            :method  => 'GET',
+            :path    => 'availability-zones'
```
Comment (Carthaca) [inline lib/fog/shared_file_system/openstack/requests/list_availability_zones.rb]: I suggest to build support for microversions prior 2.6, too with `os-availability-zone`, check with `microversion_newer_than?('2.6')`

**Review by Carthaca — CHANGES_REQUESTED** [review]

File: lib/fog/shared_file_system/openstack.rb [hunk 3]
```diff
@@ -60,6 +60,9 @@ class OpenStack < Fog::Service
       request :update_snapshot
       request :delete_snapshot
 
+      # availability zones
+      request :list_availability_zones
+
```
Comment (Carthaca) [inline lib/fog/shared_file_system/openstack.rb:67]: you are implementing the model + collection, so make them also available, please

**Comment by andypf** [comment]
@Carthaca: lgty?

**Review by Carthaca — APPROVED** [review]

**Comment by Carthaca** [comment]
lgtm

**Pull request merged**